lib/mutations/command.rb

```diff
@@ -56,8 +56,8 @@ def input_filters
     # Instance methods
     def initialize(*args)
       @raw_inputs = args.inject({}.with_indifferent_access) do |h, arg|
-        raise ArgumentError.new("All arguments must be hashes") unless arg.is_a?(Hash)
-        h.merge!(arg)
+        raise ArgumentError.new("All arguments must be hashes or respond to to_hash") unless arg.respond_to?(:to_hash)
+        h.merge!(arg.to_hash)
```

Comment:
`Hash#merge!` [automatically calls `to_hash`](https://github.com/ruby/ruby/blob/v2_5_0/hash.c#L2563) on its argument, so we don't have to do it ourselves.

Additionally, `h` here is a HashWithIndifferentAccess, which [overrides `Hash#merge!`](https://github.com/rails/rails/blob/v5.1.5/activesupport/lib/active_support/hash_with_indifferent_access.rb#L125-L130) to convert any nested hashes to HWIAs - but [falls back to the original implementation](https://github.com/rails/rails/blob/v5.1.5/activesupport/lib/active_support/hash_with_indifferent_access.rb#L123) if its argument is another HWIA. By calling `to_hash` here, we would force it to always use the slower path, even when `arg` is a HWIA.

Reply:
Good to know. Thanks for the explanation, I've removed the redundant call.